cloudevents/sdk/event/base.py

```diff
@@ -116,7 +125,11 @@ def MarshalJSON(self, data_marshaller: typing.Callable) -> typing.IO:
         return io.BytesIO(json.dumps(props).encode("utf-8"))
 
     def UnmarshalJSON(self, b: typing.IO, data_unmarshaller: typing.Callable):
-        raw_ce = json.load(b)
+        if isinstance(b, io.BytesIO) or isinstance(b, io.StringIO):
+            raw_ce = json.load(b)
+        else:
+            raw_ce = b
+
```

Comment:
What's the motivation behind the changes in this file?

Reply:
The UnmarshalJSON function only gets called during structured requests which I'll add support in a future PR. I noticed the code wanted specifically IO stream objects as data for reasons I was unsure of, however I wanted to be able to handle dictionaries as the body instead of a json stream objects so in some uncommitted version of the structured code that if statement was used. I've currently refactored it back to just simply json.load but I may play around with that in a future PR.